src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,36 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private final MachineProducerStrategy producerStrategy;
+
+    public MachineServiceImpl() {
+        producerStrategy = new MachineProducerStrategy();
+    }
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        return producerStrategy.getMachineProducerStrategy(type)
+                .map(machineProducer -> new ArrayList<Machine>(machineProducer.get()))
```

Comment:
consider creating a separate variable for `machineProducer.get()` (taking into account that `Optional` will be removed)